jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/content/PathContentSource.java

```diff
@@ -36,24 +38,39 @@
  */
 public class PathContentSource implements Content.Source
 {
+    // TODO in 12.1.x reimplement this class based on ByteChannelContentSource
+
     private final AutoLock lock = new AutoLock();
     private final SerializedInvoker invoker = new SerializedInvoker();
     private final Path path;
     private final long length;
     private final ByteBufferPool byteBufferPool;
-    private int bufferSize = 4096;
-    private boolean useDirectByteBuffers = true;
+    private int bufferSize;
+    private boolean useDirectByteBuffers;
     private SeekableByteChannel channel;
     private long totalRead;
     private Runnable demandCallback;
     private Content.Chunk errorChunk;
 
     public PathContentSource(Path path)
     {
-        this(path, null);
+        this(path, null, true, -1);
     }
 
     public PathContentSource(Path path, ByteBufferPool byteBufferPool)
+    {
+        this(path, new ByteBufferPool.Sized(byteBufferPool, true, -1));
+    }
+
+    public PathContentSource(Path path, ByteBufferPool.Sized byteBufferPool)
```

Comment:
I don't think this works as you expect, for example if you do:
```
ByteBufferPool bufferPool = new ByteBufferPool.Sized(new ArrayByteBufferPool());
new PathContentSource(path, bufferPool);
```

then it ends up in the other constructor and re-wraps and overrides the values from the sized pool. So maybe you need an `instanceOf` check for `ByteBufferPool.Sized` in the above constructor?